src/App.jsx

```diff
@@ -1,3 +1,239 @@
+      setCharCount(0); // Reset character count
+    } catch (error) {
+      console.error('Error submitting thought:', error);
+    }
+  };
+  
+  
+
+  const handleLike = async (thoughtId) => {
+    try {
+      const response = await fetch(`https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts/${thoughtId}/like`, {
+        method: 'POST',
+      });
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+      // Update the likes state to reflect the new like count
+      setLikes((prevLikes) => ({
+        ...prevLikes,
+        [thoughtId]: (prevLikes[thoughtId] || 0) + 1,
+      }));
+    } catch (error) {
+      console.error(`Error liking thought ${thoughtId}:`, error);
+    }
+  };
+
+  return (
+    <React.Fragment>
+    <div className="app-container">
+      <div className="header">
+      <h1>Project Happy Thoughts</h1>
+      <div className='marquee'>
+        <h2>Share your happy thoughts with the world!</h2>
+        </div>
+      </div>
+      <div className="thought-form">
+        <h2>What&rsquo;s making you happy right now?</h2>
+        <form>
+          <textarea
+            // value={newThought}
+            // onChange={(e) => setNewThought(e.target.value)}
+            // placeholder="Enter your thought (140 characters or less)"
+            // maxLength={140}
+            value={newThought}
+            onChange={(e) => {
+              const text = e.target.value;
+              if (text.length <= 140) {
+                setNewThought(text);
+                setCharCount(text.length);
+              }
+            }}
+            placeholder="Enter your thought (140 characters or less)"
+            maxLength={140}
+             
+          />
+          <div className="post-length">
+            <p className="error"></p>
+            <p className="length">
+              {charCount}/{140}
+            </p>
+          </div>
+          
+        
+          <button onClick={handleNewThoughtSubmit} id="submitPostBtn" aria-label="button for submitting your post"> 
+          <span className="emoji" aria-label="heart emoji">💕 Send Happy Thought</span>
+          <span className="emoji" aria-label="heart emoji">💕</span> 
+          </button>
+          
+          
+        </form>
+        
+      </div>
+      <div className="thought-list">
+        {thoughts.map((thought) => (
+          <div key={thought._id} className="thought-item">
+            <p>{thought.message}</p>
+            <div className="thought-item-bottom">
+              <div className="thought-item-bottom-left">
+                <button onClick={() => handleLike(thought._id)}>Like</button>
+                <span>Likes: {likes[thought._id] || 0}</span>
+              </div>
+            <span>Created <TimeAgo createdAt={thought.createdAt} /></span>
+            </div>
+          </div>
+        ))}
+      </div>
+    </div>
+    <Footer />
+    </React.Fragment>
+  );
+}
+
+
+
+
```

Comment:
Both for error catching and for easy builing and reading, it would have been good to split all the functions in the app into different components - one component to fetch API, one for new thoughts, one for like button etc